src/webgpu/api/operation/copyBetweenLinearDataAndTexture.spec.ts

```diff
@@ -413,20 +414,27 @@ class CopyBetweenLinearDataAndTextureTest extends GPUTest {
    * on the texture by emulating the copy behaviour here directly.
    */
   updateLinearTextureDataSubBox(
-    { bytesPerRow, rowsPerImage }: { bytesPerRow: number; rowsPerImage: number },
-    sourceDataLayout: Required<GPUTextureDataLayout>,
+    destinationDataLayout: Required<GPUImageDataLayout>,
+    sourceDataLayout: Required<GPUImageDataLayout>,
     copySize: GPUExtent3DDict,
-    origin: Required<GPUOrigin3DDict>,
+    destinationOrigin: Required<GPUOrigin3DDict>,
+    sourceOrigin: Required<GPUOrigin3DDict>,
     format: SizedTextureFormat,
     destination: Uint8Array,
     source: Uint8Array
   ): void {
-    for (const texel of this.iterateBlockRows(copySize, origin, format)) {
-      const sourceOffset = this.getTexelOffsetInBytes(sourceDataLayout, format, texel, origin);
+    for (const texel of this.iterateBlockRows(copySize, sourceOrigin, format)) {
+      const sourceOffset = this.getTexelOffsetInBytes(
+        sourceDataLayout,
+        format,
+        texel,
+        sourceOrigin
+      );
       const destinationOffset = this.getTexelOffsetInBytes(
-        { bytesPerRow, rowsPerImage, offset: 0 },
+        destinationDataLayout,
         format,
-        texel
+        texel,
+        destinationOrigin
```

Comment:
I think somehow the omission of offset and the omission of destinationOrigin were previously canceling each other out. I don't understand why these tests passed to begin with, but this seems right, and at least in Chrome, this makes the tests pass.

Reply:
I'm not sure :(